torch/csrc/jit/frontend/lexer.h

```diff
@@ -299,8 +434,196 @@ struct TORCH_API SharedParserData {
+          if (i >= kIsNotSpaceIndex + 1) {
+            // Kick out to the after-loop flow, which will correctly
+            // record that we found TK_IS.
+            break;
+          }
+          possible_special_token = '\0';
+        } else if (ch == ' ') {
+          continue;
+        }
+        if (possible_special_token && i == sizeof(kIsNot) - 2 &&
+            (!pos.has_next() || !valid_ident_char(*(pos + 1)))) {
+          *kind = TK_ISNOT;
+          *end = pos.next_iter();
+          return;
+        }
+      } else if (possible_special_token == kMaybeNotIn) {
+        if (ch != kNotIn[i]) {
+          if (i >= kNotInSpaceIndex + 1) {
+            // Kick out to the after-loop flow, which will correctly
+            // record that we found TK_NOT.
+            break;
+          }
+          possible_special_token = '\0';
+        } else if (ch == ' ') {
+          continue;
+        }
+
+        if (possible_special_token && i == sizeof(kNotIn) - 2 &&
+            (!pos.has_next() || !valid_ident_char(*(pos + 1)))) {
+          *kind = TK_NOTIN;
+          *end = pos.next_iter();
+          return;
+        }
+      }
+      if (valid_ident_char(ch)) {
+        if (i < token_chars.size()) {
+          token_chars[i] = ch;
+        }
+        continue;
+      }
+      break;
+    }
+
+    // These two possible_special_token checks have to be after the
+    // loop and not in the loop because we might see end-of-input
+    // (e.g., the entire input `not p`).
+    if (possible_special_token == kMaybeIsNot) {
+      if (i >= kIsNotSpaceIndex) {
+        *kind = TK_IS;
+        *end = start + kIsNotSpaceIndex;
+        return;
+      }
+    } else if (possible_special_token == kMaybeNotIn) {
+      if (i >= kNotInSpaceIndex) {
+        *kind = TK_NOT;
+        *end = start + kNotInSpaceIndex;
+        return;
+      }
+    }
+
+    *end = pos;
+    *kind = identTokenKind(token_chars, i);
+  }
+
+  template <size_t N>
+  static constexpr uint64_t stringToUint64(const char (&str)[N]) {
+    static_assert(N <= 9);
+    uint64_t result = 0;
+    for (auto i : c10::irange(N)) {
+      if (!str[i]) {
+        return result;
+      }
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+      result |= static_cast<uint64_t>(str[i]) << (8 * i);
+#elif __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+      result |= static_cast<uint64_t>(str[i]) << (56 - 8 * i);
+#else
+#error "Unexpected or undefined value of __BYTE_ORDER__"
+#endif
+    }
+    return result;
+  }
+
+  static int identTokenKind(
+      const std::array<char, 8>& token_chars,
+      size_t token_length) {
+    if (token_length > token_chars.size()) {
+      return TK_IDENT;
+    }
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
+    static_assert(stringToUint64("and") == 0x646e61);
+    static_assert(stringToUint64("Ellipsis") == 0x73697370696c6c45);
+#else
+    static_assert(stringToUint64("and") == 0x616e640000000000);
+    static_assert(stringToUint64("Ellipsis") == 0x456c6c6970736973);
```

Comment:
original PR had these constants wrong. (I forgot the trailing zeros on the first one, and somehow had 8083 instead of 7073 in the second) Since no big-endian builds run by default, we didn't notice until land.

Reply:
(this time we have ciflow/s390 to make sure we notice if the failure continues)